### ISSUE TITLE
Unable to edit and then upload/overwrite the file .txt

### ISSUE THREAD

**Issue by otherpaco** [haiwen/seadroid#691]
Android 7.0 (Motorola)
Seafile Server 6.2.2 (problem existed with 6.0.8, too)
App 2.1.17 - 2.1.16 - 2.1.15 - 2.1.14 all have this bug

Latest working version 2.1.13

older Versions 2.1.12 crashes on Android 7.0 - 2.1.11 works - 2.1.10 works

When I open a .txt file it opens in the editor I can edit it but when I save it, it is not saved.
On the server remains the old version.
In the app I see the old version.

I tried Google Docs and Jota+

When I open it with Jota+ it says  this

Opened via Content Provider. So you can't overwrite this file.

I have the same problem but no message in google docs.

Maybe some methods of Content Provider are not implemented correctly.

No problems as webservice or Linuxapp. No problem on the phone to download or upload pictures.

**Comment by otherpaco** [comment]
Bug still exists in Version 2.1.18

**Comment by freeplant** [comment]
Thanks for reporting. We will try to reproduce the problem later.

**Comment by otherpaco** [comment]
@freeplant Do you need further instructions on reproducing this? It is still present in the latest release (2.1.20). This problem basically completely prevents users from editing files with other installed apps.

**Comment by dbet1** [comment]
I also have this problem on a newly installed android 7 device. If I open an excel sheet, it will be showed in the corresponding app (planmaker for me). But the title there is shown as "unknown2 instead of the filename. I can change the file, but if I tap save, I have to choose a folder and a filename, because it is a "new file".
On my devices with android 4.4 and 6.0.1 with the same document seafile opens it with the planmaker app, which shows me in the title bar the right document name. If I change something and tap save, seafile uploads the changed file.

**Comment by otherpaco** [comment]
still present in the latest release (2.2.0)

as workaround untill it is fixed:

Deintsall your Seafile on Android
Get version 2.1.13 from this git here https://github.com/haiwen/seadroid/releases/tag/2.1.13
Install that .apk and everything should work fine.

**Comment by freeplant** [comment]
Actually, we can't reproduce the problem. But we will try a possible solution it in the next release.

**Comment by otherpaco** [comment]
Hello, just tried it with a different seafile server - same problems
tried it with .odt file - same problems

If you need more info or I can be helpful in another way just let me know, otherwise I wait and hope for 2.2.1

Thank you

**Comment by otherpaco** [comment]
Sorry to say so but same problem with 2.2.1
Maybe @lins05 can help here as he released the latest working version (2.1.13)
If you need more info to reproduce the problem let me know.

Thank you

**Comment by dbet1** [comment]
Interesting: with Planmaker Mobile the file opens as Unknown, a new file. With Planmaker HD it opens as the file with the data. But after changing something and saving it, seafile doesn't upload the file.

**Comment by fhill122** [comment]
Exact same issue here.
Files opened in other apps are all read only after version 2.1.13

**Comment by dbet1** [comment]
I have reinstalled version 2.1.13 an now I can open files and after changing it, seafile uploads it.

**Comment by dbet1** [comment]
After a few days using version 2.1.13 I see that in the transfer list a downloaded or uploaded file never disappears. They steys for ever in the list. Confusing.
Why does not anyone fix the mistake? It must be possible to find out where the problem lies.

**Comment by dbet1** [comment]
I have tested version 2.2.1 and 2.2.4 with two devices which have two different Android versions. The test scenario was to open an encrypted library, then open a file, change something in this file and save the file. After that I checked that it is uploaded to the library. The same I tested with unencrypted libraries.

*** Version 2.2.1 ***

Samsung Galaxy Note 8.0 mit Android 4.4.4
- Download of documents, the change and upload are OK. It works with encrypted and unencrypted libraries.

Samsung Galaxy Tab S3 mit Android 8.0.0
- Documents were downloaded as new file, I have to save it local, then uploade to seafile. This is unusable.


*** Version 2.2.4 ***

Samsung Galaxy Note 8.0 mit Android 4.4.4
- Documents are downloaded, but after a change they cannot be upload. The error message is not helpful: Upload failed.
- During a second try there was an app abort during download. After that, I can't download any file in an encrypted library. I had to change to "encrypt data on the server", after that I could download the file again, but upload also failes.
- Documents in unencrypted libraries can be downloaded but also not uploaded: Upload failed.

Samsung Galaxy Tab S3 mit Andorid 8.0.0
- Documents were downloaded as new file, I have to save it local, then uploade to seafile. Tested only with unencrypted libraries because passwords for encrypted libraries were not accepted at all.

*** Conclusion ***
Because the Note 8.0 I don't use anymore I have still to stay with version 2.1.13. With this version the Dowload/Upload is correctly also on the Tab S3.

Seafile is a wonderful piece of software, but the android app is a pain.

**Comment by dbet1** [comment]
I have upgraded my server to 6.3.2. Now the situation is as follows with seadroid 2.2.6:

Samsung Galaxy Note 8.0 mit Android 4.4.4
- Documents can be downloaded, modified and uploaded, whether the library is encrypted or not.

Samsung Galaxy Note 8 mit Andorid 8.0.0
- Documents can be downloaded and opened in the corresponding app. But then you realize that it is a new file, even if the content is there. Depending on the app, you can not change anything or you will be asked for a file name when saving. Just as if it were a new document.
- But you can open the file via a file manager from the local Seafile cache, modify and save. Afterwards you go back to the app Seafile and select the upload function on the corresponding file. After that, the modified version is saved saved in the Seafile cloud.

**Comment by otherpaco** [comment]
Hello,

I just updated to 2.2.8 and now the problem is kind of solved.
The .txt files open in the build in editor and even .md files work pretty fine.

I would say solved even though it seems some people still have problems with up- and download of files but I think that should be covered in another issue.

Thank you

**Comment by dbet1** [comment]
No, with 2.2.8 the upload problem is still there on my both devices.

Earlier it was possible to open a file from the seafile cache and if I have modified it, seafile has uploaded it after starting the app. Now it gives only an error.

How can I debug this error?

**Comment by otherpaco** [comment]
I agree with dbet1, it works for .txt .md and maybe some other but you still can not open files in other apps and save them to your seafile. e.g. .ods spreadsheets.

**Comment by otherpaco** [comment]
Sorry missclicked STILL OPEN

**Comment by otherpaco** [comment]
I agree with dbet1, it works for .txt .md and maybe some other but you still can not open files in other apps and save them to your seafile. e.g. .ods spreadsheets.

**Comment by dbet1** [comment]
Since four month the problem is unchanged. Still I have to save the document somewhere on my Android device, then upload to my Seafile cloud. Lousy.

**Comment by t-m-w** [comment]
I can verify this issue here in Android 9.0 (LineageOS 16). In the past, opening the Seafile Android app and refreshing would remind it to upload any changes that had occurred in the local copy. Not anymore.

If the app supports Android's standard system file picker, you can try browsing for a file, swiping open the lefthand sidebar, and choosing Seafile to browse to it. I used to have more success that way. (Right now, for me, choosing Seafile in the system file browser causes the browser to crash, so I can't verify this will help, but that's another issue.)

I have one last suggestion which may not help for encrypted libraries, but for now you might be able to work around this by using Seafile's WebDAV interface, if you have it set up and use an app that supports WebDAV. That's what I did with the password manager I was using, which thankfully supported this natively. There are also other apps that can synchronize to WebDAV and maybe show such sites as content providers, but I don't have much familiarity with those.

**Comment by dbet1** [comment]
I never have seen that the Android system file picker supports Seafile. I am using Samsung devices with Android 8. As I almost use encrypted libraries, WebDAV is not a solution for me.

**Comment by florencebiree** [comment]
Same problem here, with the last Seadroid from F-Droid, and lineageOS 17.1.

The 2.1.23 version works well…

**Comment by DiTsi** [comment]
Same problem. Thanks @florencebiree: [2.1.20 version](https://github.com/haiwen/seadroid/releases/tag/2.1.20) with external editor works well
